be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -412,7 +423,7 @@ Status BetaRowsetWriter::_load_noncompacted_segments(
  *     length is beyond (config::segcompaction_threshold_segment_num / 2)
  */
 Status BetaRowsetWriter::_find_longest_consecutive_small_segment(
-        SegCompactionCandidatesSharedPtr segments) {
+        SegCompactionCandidatesSharedPtr segments, size_t small_threshold, bool is_greedy) {
```

Comment:
I think it's better than _find_longest_consecutive_small_segment just find rowset with 2 set, such as rename_segments_set and compact_segment_set, instead of doing rename inside this funtion

Reply:
I agree it is a little mess in the find function... but moving rename out dose not seems to help with that (and may bring additional loops & iterator stuff than rename inplace). I should instead, refine the description comments for the function.